### PR TITLE
Reimplement queryInternal with a new inner method

### PR DESCRIPTION
**Pull request by j-calvert**
Reimplement queryInternal with a new method, wrapped by the older one, that obtains column type info (in preparation for further changes for https://github.com/facebook/osquery/issues/4799 ).

This is a partial replacement for (and part of the breaking up of) this [larger pull request](https://github.com/facebook/osquery/pull/4827).

I've incorporated the suggestions made in that pull request, for the relevant parts of the code.

### CODE REVIEW

File: osquery/sql/sqlite_util.cpp [hunk 1]
```diff
@@ -459,41 +459,115 @@ Status QueryPlanner::applyTypes(TableColumns& columns) {
...
+
+  /* The big while loop.  One iteration per statement */
+  while ((sql[0] != '\0') && (SQLITE_OK == rc)) {
+    const auto lock = instance->attachLock();
+
+    rc = sqlite3_prepare_v2(
+        instance->db(), sql, -1, &prepared_statement, &leftover_sql);
+    if (rc != SQLITE_OK) {
+      Status s = Status::failure(sqlite3_errmsg(instance->db()));
+      if (prepared_statement != nullptr) {
```
Comment (fmanco) [inline osquery/sql/sqlite_util.cpp]: No need to check for `nullptr` since `sqlite3_finalize()` can be called with `nullptr`.

File: osquery/sql/sqlite_util.cpp [hunk 2]
```diff
@@ -459,41 +459,115 @@ Status QueryPlanner::applyTypes(TableColumns& columns) {
...
+          // obtained/conveyed as text/string
+          row[colNames[i]] = std::string(reinterpret_cast<const char*>(
+              sqlite3_column_text(prepared_statement, i)));
+        }
+      }
+      results.push_back(std::move(row));
+      rc = sqlite3_step(prepared_statement);
+    } while (SQLITE_ROW == rc);
+  }
+  if (!(rc == SQLITE_OK || rc == SQLITE_DONE)) {
```
Comment (fmanco) [inline osquery/sql/sqlite_util.cpp]: I think you should always get here with a `SQLITE_DONE` code and `sqlite3_step()` doesn't return `SQLITE_OK`. But since the documentation isn't very clear please test it for the edge case of no results.

**Comment by fmanco** [comment]
ok to test

**Comment by j-calvert** [comment]
Squashed and rebased.

**Comment by Smjert** [comment]
Closing since this already landed via this (direct, no PR) commit https://github.com/osquery/osquery/commit/5461a3b79def92e10a408393491cd3c9573be800, which is more recent than this PR.
